### PR TITLE
DOC: documents about K8s deployment

### PR DESCRIPTION
**Pull request by ChengjieLi28**
Add documents about k8s deploy.

## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass

### CODE REVIEW

File: doc/source/deployment/kubernetes.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+=====================
+Kubernetes deployment
+=====================
+
+Prerequisites
+-------------
+Install Xorbits on the machine where you plan to run the kubernetes deploy code.
+Refer to :ref:`installation document <installation>`.
+
+Currently we support Xorbits kubernetes cluster deployment based on Minikube.
```
Comment (qianduoduo0904) [inline doc/source/deployment/kubernetes.rst:12]: We can provide the doc about deploying to Kubernetes directly.

File: doc/source/deployment/cloud.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+Prerequisites
+-------------
+Currently we support deploying Xorbits to Amazon EKS cluster.
+
+Install Xorbits on the machine where you can control your Amazon EKS cluster.
+Refer to :ref:`installation document <installation>`.
+
+Amazon EKS
+----------
+Firstly, make sure you have an EKS cluster and it can access `our Dockerhub <https://hub.docker.com/repository/docker/xprobe/xorbits>`_ to pull the Xorbits image.
```
Comment (UranusSeven) [inline doc/source/deployment/cloud.rst:16]:
```suggestion
Firstly, make sure your EKS cluster can access `our Dockerhub <https://hub.docker.com/repository/docker/xprobe/xorbits>`_.
```

File: doc/source/deployment/cloud.rst [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    from kubernetes import config
+    from xorbits.deploy.kubernetes import new_cluster
+    cluster = new_cluster(config.new_client_from_config(), worker_cpu=1, worker_mem='4g', cluster_type='eks')
+
+
+Note that the option ``cluster_type='eks'`` cannot be ignored.
+
+When a log of the form ``Xorbits endpoint http://<ingress_service_ip>:80 is ready!`` appears,
+you can access the web page of your xorbits cluster through the endpoint in the log.
```
Comment (UranusSeven) [inline doc/source/deployment/cloud.rst:32]:
```suggestion
you can access the web UI of your Xorbits cluster using the endpoint.
```

File: doc/source/deployment/cloud.rst [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+.. code-block:: python
+
+    from kubernetes import config
+    from xorbits.deploy.kubernetes import new_cluster
+    cluster = new_cluster(config.new_client_from_config(), worker_cpu=1, worker_mem='4g', cluster_type='eks')
+
+
+Note that the option ``cluster_type='eks'`` cannot be ignored.
+
+When a log of the form ``Xorbits endpoint http://<ingress_service_ip>:80 is ready!`` appears,
```
Comment (UranusSeven) [inline doc/source/deployment/cloud.rst:31]:
```suggestion
You should be able to see ``Xorbits endpoint http://<ingress_service_ip>:80 is ready!`` soon, and
```

File: doc/source/deployment/kubernetes.rst [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+------------
+By default, the image tagged by ``xprobe/xorbits:<xorbits version>`` on `our Dockerhub <https://hub.docker.com/repository/docker/xprobe/xorbits>`_
+is used in the kubernetes deployment. Each released version of Xorbits has its image, distinguished by the ``<xorbits version>``.
+
+If you need to build an image from source, the related Dockerfiles exists at `this position <https://github.com/xprobe-inc/xorbits/tree/main/python/xorbits/deploy/docker>`_ for reference.
+You can follow the `Docker document <https://docs.docker.com/engine/reference/commandline/build/>`_ to build your own Xorbits image.
+
+After you build your own image, push it to a image repository accessible by your K8s cluster, e.g. your own DockerHub namespace.
+
+Finally, specify your own image during the deployment process through the ``image`` option of the ``new_cluster`` api.
```
Comment (qianduoduo0904) [inline doc/source/deployment/kubernetes.rst:72]:
```suggestion
Finally, specify your own image during the deployment process through the ``image`` option of the :meth:`xorbits.deploy.kubernetes.client.new_cluster` api.
```

File: doc/source/deployment/cloud.rst [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+Then, deploy Xorbits cluster, for example:
+
+.. code-block:: python
+
+    from kubernetes import config
+    from xorbits.deploy.kubernetes import new_cluster
+    cluster = new_cluster(config.new_client_from_config(), worker_cpu=1, worker_mem='4g')
+
+
+Note that the option ``cluster_type`` of the ``new_cluster`` api equals to ``auto`` by default, which means that
```
Comment (UranusSeven) [inline doc/source/deployment/cloud.rst:29]:
```suggestion
Note that the option ``cluster_type`` of the function ``new_cluster`` has default value ``auto``, which means that
```

File: doc/source/deployment/cloud.rst [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+the ``kubectl`` context on your machine will be detected. Please make sure kubectl uses your eks context.
```
Comment (UranusSeven) [inline doc/source/deployment/cloud.rst:30]:
```suggestion
Xorbits will detect the ``kubectl`` context automatically. Please make sure ``kubectl`` is using the correct EKS context.
```

**Review by qianduoduo0904 — APPROVED** [review]
LGTM

**Review by UranusSeven — APPROVED** [review]

**Pull request merged**